plugins/Logs/js/logparse.js

```diff
@@ -0,0 +1,48 @@
+import fs from 'fs'
+import { readdirRecursive } from './utils.js'
+
+const siadir = SiaAPI.config.siad.datadir
+
+const fileSize = (logpath) =>
+	fs.statSync(logpath).size
+
+// cleanLog removes partial lines.
+export const cleanLog = (logtext) =>
+	logtext.slice(logtext.indexOf('\n') + 1, logtext.lastIndexOf('\n') + 1)
+
+// readLog takes a log path and returns a string representing the log data.
+export const readLog = (logpath, nbytes) => {
+	const len = nbytes || fileSize(logpath)
+	const startPos = Math.max(0, fileSize(logpath) - len)
+	const buf = new Buffer(len)
+	const fd = fs.openSync(logpath, 'r')
+	fs.readSync(fd, buf, 0, len, startPos)
+
+	return cleanLog(buf.toString())
```

Comment:
might be better to slice the buffer instead of the string. I doubt there's much performance impact though